pkg/minio/minio.go

```diff
@@ -61,15 +65,28 @@ func NewClient(bucket *sourcev1.Bucket, secret *corev1.Secret, tlsConfig *tls.Co
 		opt.Creds = credentials.NewIAM("")
 	}
 
+	var transportOpts []func(*http.Transport)
+
 	if opt.Secure && tlsConfig != nil {
-		// Use the default minio transport, but override the TLS config.
-		secure := false // true causes the TLS config to be defined internally, but here we have our own so we just pass false.
-		transport, err := minio.DefaultTransport(secure)
+		transportOpts = append(transportOpts, func(t *http.Transport) {
+			t.TLSClientConfig = tlsConfig.Clone()
+		})
+	}
+
+	if proxyURL != nil {
+		transportOpts = append(transportOpts, func(t *http.Transport) {
+			t.Proxy = http.ProxyURL(proxyURL)
+		})
+	}
+
+	if len(transportOpts) > 0 {
+		transport, err := minio.DefaultTransport(true /*secure*/)
```

Comment:
> Maybe use opt.Secure now that this will run regardless of the provided tls config.
If I read this right https://github.com/minio/minio-go/blob/v7.0.70/transport.go#L73, presence of SSL_CERT_FILE environment variable will configure TLS even if the Bucket object has .spec.insecure set to true. No TLS config but just proxy config will create a non-zero transportOpts and run the above code to configure TLS.

:+1: fixed, thanks!

> Maybe also remove the above if condition so that default transport can be constructed using SSL_CERT_FILE from environment in absence of TLS config or proxy option.

The MinIO library already does that (it's the "default" transport after all):

https://github.com/minio/minio-go/blob/v7.0.70/api.go#L228-L234